tachyon/zk/plonk/lookup/permute_expression_pair.h

```diff
@@ -0,0 +1,167 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_LOOKUP_PERMUTE_EXPRESSION_PAIR_H_
+#define TACHYON_ZK_PLONK_LOOKUP_PERMUTE_EXPRESSION_PAIR_H_
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include "absl/container/btree_map.h"
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/zk/plonk/error.h"
+
+namespace tachyon::zk {
+
+template <typename PCSTy>
+struct ExpressionPair {
+  using Evals = typename PCSTy::Evals;
```

Comment:
I want them to be templaized with a minimum type requirement. 

```suggestion
template <typename Evals>
struct ExpressionPair {
```